### ISSUE TITLE
The `frame count` has decreased compared to the `none_count`

### ISSUE THREAD

**Issue by WK-GiHu** [JohannesAppel/johannesappel-#20]
@WK-GiHu 
The `frame count` has decreased compared to the `none_count`, when the `none count` was almost zero and the `frame count` was almost the same as the `max frames`
```
canlib version: 8.30
Using channel: Kvaser Leaf Professional HS (channel 0), EAN: 73-30130-00404-7
CanBus.stop()
EXIT, frame_count:101, none_count:899, trashed_count:0
```

_Originally posted by @JohannesAppel in https://github.com/JohannesAppel/johannesappel-/issues/19#issuecomment-574134337_

**Comment by WK-GiHu** [comment]
@JohannesAppel 

>The frame count has decreased compared to the none_count

Should be equal to `max_frames`, a low `frame_count` and therefore a high `none_count` indicates:
1. Hardware Issue, no Frames get read from **CANBUS**
2. Threads are unbalanced, the **consumer**, `.get()` graps more timeslices than the **producer** `.put(...)`.

>when the none count was almost zero and the frame count was almost the same as the max frames

This is the desired behavior.
I didn't get your sentence meaning?

**Comment by JohannesAppel** [comment]
@WK-GiHu 
Yes, that is the desired behavior. We want our `frame count` as close as possible to the `max frames`.

**Comment by WK-GiHu** [comment]
@JohannesAppel 

As it stands, we could not distinguish between, `None` Frames from ` Kvaser.read(...` and `None` Frames from `CanBus.get()`. We need some sort of ***TRISTATE***?

We should switch from `max_frames = 1000` to `timer = 60 seconds` to count how many Frames are `.get()` within a timeframe.
I ask again: ***Did you know how many Frame get fired by the Hardware within a timeframe?***

Maybe you can restict to **one** Frame by *frame.id*?
We are back to the question: **Is it realtime or not?**

**Comment by JohannesAppel** [comment]
@WK-GiHu 
Hi, sorry for replying only now. Have been busy with another project the last few days. My time here at the business that I'm doing my vacation work is coming to an end. Tomorrow is my last day. So I won't be able to continue the project. Sorry for only leaving it half way. I appreciate your help a lot, I've learned a lot from you the last month and a half. 

To give you an update about the project. So in found a easier way for doing the project without having a need to see the values real time. I read a few thousand lines and then close the line so that the build up won't affect the output, and that's only to get a base line so that later on I can compare that data to the data after I've triggered a control unit on the bike. And it worked, but I don't see the data being outputted at all which is sad because that would've been nice if it was possible to see it. 

I also came to the conclusion after this month and a half that the problem might be that the `Kvaser` has some sort of build up when reading the data and sending it to the computer, because the longer the code runs the slower the output gets, meaning that somewhere there is a build up and then a slower data transfer. So either Python or the `Kvaser` is the problem. 

Again, thank you for your help. Hope to work with you again in the future and learn more from you.
